pkg/cluster/clustermsi.go

```diff
@@ -143,3 +148,44 @@ func (m *manager) clusterMsiSecretName() (string, error) {
 
 	return fmt.Sprintf("%s-%s", m.doc.ID, clusterMsi.Name), nil
 }
+
+func (m *manager) clusterIdentityIDs(ctx context.Context) error {
+	clusterMsiResourceId, err := m.doc.OpenShiftCluster.ClusterMsiResourceId()
+	if err != nil {
+		return err
+	}
+
+	uaMsiRequest := dataplane.UserAssignedMSIRequest{
+		IdentityURL: m.doc.OpenShiftCluster.Identity.IdentityURL,
+		ResourceIDs: []string{clusterMsiResourceId.String()},
+		TenantID:    m.doc.OpenShiftCluster.Identity.TenantID,
+	}
+
+	msiCredObj, err := m.msiDataplane.GetUserAssignedIdentities(ctx, uaMsiRequest)
+	if err != nil {
+		return err
+	}
+
+	if msiCredObj.CredentialsObject.ExplicitIdentities == nil ||
+		len(msiCredObj.CredentialsObject.ExplicitIdentities) == 0 ||
+		msiCredObj.CredentialsObject.ExplicitIdentities[0] == nil ||
+		msiCredObj.CredentialsObject.ExplicitIdentities[0].ClientID == nil ||
+		msiCredObj.CredentialsObject.ExplicitIdentities[0].ObjectID == nil {
+		return errClusterMsiNotPresentInResponse
```

Comment:
We are checking the CredentialsObject also for the federatedCredentialsClient, is there a scope to have a reusable function?

Also, can we have separate error message for MSI not found and ClientID/ObjectID missing.

Reply:
Done.